packages/server/src/course/course.controller.ts

```diff
@@ -301,6 +304,40 @@ export class CourseController {
     await UserCourseModel.remove(userCourse);
   }
 
+  @Post('submit_course')
+  async submitCourse(@Body() body: SubmitCourseParams): Promise<void> {
+    const season = body.semester.split(' ')[0];
+    const year = parseInt(body.semester.split(' ')[1]);
+
+    const semester = await SemesterModel.findOne({
+      where: { season, year },
+    });
+    if (!semester)
+      throw new BadRequestException(
+        ERROR_MESSAGES.courseController.noSemesterFound,
+      );
+
+    // create the submitted course
+    const course = await CourseModel.create({
+      name: body.name,
+      coordinator_email: body.coordinator_email,
+      icalURL: body.icalURL,
+      semesterId: semester.id,
+      enabled: false,
+      pending: true,
+      timezone: body.timezone,
+    }).save();
+
+    // create CourseSectionMappings for each submitted section number
+    body.sections.forEach(async (section) => {
+      await CourseSectionMappingModel.create({
+        genericCourseName: body.name,
```

Comment:
Do we care at all about normalizing course names from the form input (eg. cs3000 vs CS 3000 vs Algorithms and Data)? It seems like the user-entered course name gets also saved as this `genericCourseName` - which is supposed to match with what Khoury Admin gives, right?